### ISSUE TITLE
Column selection by default

### ISSUE THREAD

**Issue by francoisschwarzentruber** [ajaxorg/ace#5281]
### Describe the feature

What about having an option for having column selection by default? By "column selection", I mean the way the ace editor selects when you press Alt.

### Use Case

The typical use would be an Ascii editor made up with ace.

### Proposed Solution

An option called `selectionBehavior` which may have the following values:
- `normal`: the current selection behavior (normal line selection and column selection when pressing Alt)
- `line`: the selection is always the traditional line selection, it will disable the column selection by pressing Alt
- `column`: the selection is always the column selection (this is what I want)
- `none`: impossible to select

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### ACE version used

latest

**Comment by nightwing** [comment]
Wouldn't you also need virtual selection, that is selection going beyond the line length, for this to work as ascii editor?

**Comment by francoisschwarzentruber** [comment]
Good point @nightwing! Your question is very interesting. I think the point you raise is independent to the selection behavior. As far as I remember, in the code editor of Delphi (I used Delphi in another life ;) ), we can set the text cursor anywhere. In other words, we did not to have to add spaces or tabs, but click anywhere to write code.

So why not having another option `cursorPositioning` (a better name could be chosen) with values:
`normal` for a normal editor
`virtual` for allowing to move the cursor after the content of the line, editing will automatically add the spaces 

So:
- with `cursorPositioning` to `normal`, and `selectionBehavior` to `normal`, we have the normal behavior of an editor.
- with `cursorPositioning` to `virtual`, and `selectionBehavior` to `normal` (or `line` I do not remember, maybe Delphi did not allow for column selection at all), we have the behavior of the editor of Delphi (at least, if I remember correctly)
- with `cursorPositioning` to `virtual` and `selectionBehavior` to `column` we have the behavior of an asciiart editor.

**Comment by francoisschwarzentruber** [comment]
I actually see that the notion of virtual space (aka cursor after EOL)  (having `cursorPositioning` to `virtual`) has been discussed and apparently implemented here: https://github.com/ajaxorg/ace/issues/2579
